src/libraries/System.Threading/tests/MonitorTests.cs

```diff
@@ -447,6 +447,7 @@ public static void WaitTest()
                 Monitor.Pulse(obj);
             }
             Monitor.Exit(obj);
+            t.Join(500);
```

Comment:
Should we guard it for wasm only?

Reply:
What could go wrong doing it everywhere ?